src/net_processing.cpp

```diff
@@ -105,7 +105,8 @@ namespace {
         bool fValidatedHeaders;                                  //!< Whether this block has validated headers at the time of request.
         std::unique_ptr<PartiallyDownloadedBlock> partialBlock;  //!< Optional, used for CMPCTBLOCK downloads
     };
-    map<uint256, pair<NodeId, list<QueuedBlock>::iterator> > mapBlocksInFlight;
+    typedef std::multimap<uint256, pair<NodeId, list<QueuedBlock>::iterator>> BlockDownloadMap;
```

Comment:
I'm unsure about this change.  In particular there are a lot of mmapBlocksInFlight.count(hash) calls that would get a bit less clear...   I'll think about it some more